src/web/services/auth.service.ts

```diff
@@ -0,0 +1,28 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { environment } from '../environments/environment';
+
+/**
+ * Handles user authentication.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+
+  private backendUrl: string = environment.backendUrl;
+  private frontendUrl: string = environment.frontendUrl;
+  private withCredentials: boolean = environment.withCredentials;
+
+  constructor(private httpClient: HttpClient) {}
+
+  /**
+   * Gets the user authentication information.
+   */
+  getAuthUser(): Observable<any> {
+    const params: HttpParams = new HttpParams().set('frontendUrl', this.frontendUrl);
+    return this.httpClient.get(`${this.backendUrl}/auth`, { params, withCredentials: this.withCredentials });
```

Comment:
Should we have a protocol (instead of `any`) between the backend and frontend for the format of the data transferred? Not sure what is the practice in industry, do they enforce that?

Reply:
Ideal situation: it needs to be done.
Real situation: I can't tell how much it will be of developer inconvenience, and neither can I tell how much benefit it will bring right now. I'll keep that option open, but as far as this PR is concerned, I don't want to go down that path just yet.